framework/src/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -366,6 +368,17 @@ public AnnotatedTypeFactory(BaseTypeChecker checker) {
         }
     }
 
+    /**
+     * Issue an error if any of the support qualifiers has @Target that contains something besides
+     * TYPE_USE or TYPE_PARAMETER.
+     */
+    private void checkSupportedQuals() {
+        for (Class<? extends Annotation> annotationClass : supportedQuals) {
+            AnnotatedTypes.hasTypeQualifierElementTypes(
```

Comment:
I think it's worth a comment to explain why a method with boolean return type is being called for side effect.